src/types.rs

```diff
@@ -300,3 +300,62 @@ pub struct Metric {
     pub totalDebtCagr5Y: Option<f64>,
     pub yearToDatePriceReturnDaily: Option<f64>,
 }
+
+// TODO: Is there a macro we could make to automate From/Display traits?
```

Comment:
This would have to be a proc macro if we wanted to do something like this, I mean this would be a whole nother crate

Reply:
Ahh, damn. Well maybe we can figure something out if it becomes a pain.

Reply:
Yeah, I mean it does become painful for any application so I will consider building such a crate

It needs to be a proc macro as we are accessing the fields of the struct and looping over them and generating a `impl Display/From` which would require looking at the stream of tokens.

Reply:
It's possible that something like this already exists, but I didn't look too hard. Mabye [strum](https://crates.io/crates/strum).